widget/form.go

```diff
@@ -88,28 +126,18 @@ func (f *Form) CreateRenderer() fyne.WidgetRenderer {
 		f.itemGrid.AddObject(item.Widget)
 	}
 
-	if f.OnCancel == nil && f.OnSubmit == nil {
-		return cache.Renderer(NewVBox(f.itemGrid))
-	}
-
-	buttons := NewHBox(layout.NewSpacer())
-	if f.OnCancel != nil {
-		if f.CancelText == "" {
-			f.CancelText = "Cancel"
-		}
-
-		buttons.Append(NewButtonWithIcon(f.CancelText, theme.CancelIcon(), f.OnCancel))
-	}
-	if f.OnSubmit != nil {
-		if f.SubmitText == "" {
-			f.SubmitText = "Submit"
-		}
-
-		submitButton := NewButtonWithIcon(f.SubmitText, theme.ConfirmIcon(), f.OnSubmit)
-		submitButton.Style = PrimaryButton
-		buttons.Append(submitButton)
-	}
-	return cache.Renderer(NewVBox(f.itemGrid, buttons))
+	// Create the buttons and hide the box till they are needed
+	f.cancelButton = NewButtonWithIcon("", theme.CancelIcon(), f.OnCancel)
+	f.submitButton = NewButtonWithIcon("", theme.ConfirmIcon(), f.OnSubmit)
+	f.submitButton.Hide()
+	f.cancelButton.Hide()
+	f.buttonBox = NewHBox(layout.NewSpacer(), f.cancelButton, f.submitButton)
+	f.buttonBox.Hide()
```

Comment:
All three `Hide` calls will also be done by `setButtons`. I suggest to remove them here.